### DISCUSSION TITLE
--reload_dirs option does not behave as documented

### DISCUSSION THREAD

**Discussion by ImogenBits** [Kludex/uvicorn#1783] [Potential Issue]
The docs describe the option like this (emphathis mine):
> Specify which directories to watch for python file changes. May be used multiple times. **If unused, then by default the whole current directory will be watched.** If you are running programmatically use reload_dirs=[] and pass a list of strings.

But this is not how it actually behaves with when using the watchfile optional dependency. Currently it instead always watches the current directory in addition to whatever is specified by `reload_dirs`.  
The cli output also is wrong since it only lists the directories passed to `reload_dirs`.

This seems to be caused by [this](https://github.com/encode/uvicorn/blob/95dcf87f8971a10ff8ae3e78317f78d3d49b4611/uvicorn/supervisors/watchfilesreload.py#L71) bit of code. Since it adds the cwd into the watched directories after the config setup it also won't actually appear anywhere visible to the end user such as the cli output.

**Comment by val-melev** [comment]
It would be really nice to remove the piece of code @ImogenBits mentioned as it leads to log spam when the logging goes to a file in the working directory.

**Comment by sswam** [comment]
I only noticed this because it's causing me to run out of inotify handles. I guess I need to chdir into the directory rather than relying on the --app-dir and --reload-dir options, which do not work as advertised.
